lib/hackmode/p2p-backend-hackmode-peer.js

```diff
@@ -302,8 +302,14 @@ class P2pBackendHackmodePeer extends AsyncClass {
 
     switch (uri.protocol) {
       case 'wss:':
-        return this._ensureConnection(
-          peerInfo.peerTransport + '?a=' + peerInfo.peerAddress)
+        if (peerInfo.peerAddress === this.getId()) {
+          // Don't try to connect self
+          return Promise.resolve()
```

Comment:
- async functions wrap returns in promises, so you can just `return`
- the caller is expecting a cid to be returned, I think we might need to do this check elsewhere.